src/main/kotlin/com/rooplor/classcraftbackend/services/FormSubmissionService.kt

```diff
@@ -0,0 +1,76 @@
+package com.rooplor.classcraftbackend.services
+
+import com.opencsv.CSVWriter
+import com.rooplor.classcraftbackend.entities.FormSubmission
+import com.rooplor.classcraftbackend.messages.ErrorMessages
+import com.rooplor.classcraftbackend.repositories.FormSubmissionRepository
+import org.springframework.stereotype.Service
+import java.io.StringWriter
+
+@Service
+class FormSubmissionService(
+    private val formSubmissionRepository: FormSubmissionRepository,
+    private val formService: FormService,
+    private val authService: AuthService,
+) {
+    fun submitForm(formSubmission: FormSubmission): FormSubmission {
+        val userId = authService.getAuthenticatedUserDetails()?.id ?: throw Exception(ErrorMessages.USER_NOT_FOUND)
```

Comment:
Just in case, Now I think we have quite of feature that call userId from AuthenticatedUserDetail. Is possible to throw error when user not found directly in authService.getAuthenticatedUserDetails() to reduce duplicate code?